lib/silo/cli.rb

```diff
@@ -117,5 +117,12 @@ def cli_syntax(command, args_str = nil)
       c.action Commands, :file_pull
       c.slop.bool "-r", "--recursive", "Pull a directory and all contents"
     end
+
+    command 'file push' do |c|
+      cli_syntax(c, 'SOURCE [REPO:DEST]')
+      c.description = "Upload a file from this machine to a silo"
+      c.action Commands, :file_push
+      c.slop.bool "-r", "--recursive", "Pull a directory and all contents"
+    end
```

Comment:
I think it'd be worth having the creation of parent directories be an option which needs to be specified. If a user tries to do something like `file push pic.png pctures/2023/april/holiday` (assuming they have a directory called `pictures`) then the typo will cause a whole new tree of directories to be created without the user necessarily being aware. Maybe worth checking with @ColonelPanicks if it's necessary, you could make a good case for leaving it up to the user, but I feel like it's a nice feature to have.

Reply:
What definitely needs to happen is a change in that `--recursive` option description!

Reply:
The way `rsync` handles directories not existing is it will create one level of missing directories. I think the best aim is to replicate this behaviour however the creation of multiple levels of subdirectories isn't necessarily bad in my eyes 

```
[root@controller1 ~]# file openflighthpc.png
openflighthpc.png: PNG image data, 658 x 298, 8-bit/color RGBA, non-interlaced
[root@controller1 ~]# ls pictures/
[root@controller1 ~]# rsync -auv openflighthpc.png pictures/2023/april/holiday/
sending incremental file list
rsync: mkdir "/root/pictures/2023/april/holiday" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at main.c(663) [Receiver=3.1.3]
[root@controller1 ~]# rsync -auv openflighthpc.png pictures/2023/
sending incremental file list
created directory pictures/2023
openflighthpc.png

sent 29,326 bytes  received 71 bytes  58,794.00 bytes/sec
total size is 29,229  speedup is 0.99
[root@controller1 ~]# ls pictures/
2023
[root@controller1 ~]# ls pictures/2023/
openflighthpc.png
```

Reply:
I'll take a brief look at how complicated it would be to implement this; something's better than nothing.